src/hardware/encoder.cpp

```diff
@@ -110,13 +111,15 @@ void readMultipleEncoderRegisters(uint16_t registerAddress, uint16_t data[]) {
 
     // Send 0xFFFF (like BTT code), this returns the wanted value
     // Array length is doubled as we're using 8 bit values instead of 16
-    uint16_t arrayLength = (sizeof(data) / sizeof(data[0])) * 2;
-    uint8_t tx2buf[arrayLength] = { 0xFF };
-    uint8_t rx2buf[arrayLength] = { 0x00 };
-    HAL_SPI_TransmitReceive(&spiConfig, tx2buf, rx2buf, arrayLength, 100);
+    for (uint8_t i = 0; i < dataLength * 2; i++) {
```

Comment:
We have to use a loop here because `uint8_t tx2buf[arrayLength] = { 0xFF };` only sets the first element.